### PR TITLE
Display Monitor Tags on Status Page

### PR DESCRIPTION
**Pull request by Fallstop**
# Description

Displays the tags from the dashboard on the public status page, this makes the internal dashboard and public status to be more consistent. For example we use the tagging system to show which services depend on which, and which team is responsible for the service.

## Type of change

- User Interface
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and test it
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots
Desktop Testing:
Desktop Testing
Mobile Testing:
Mobile Testing

### CODE REVIEW

**Comment by Fallstop** [comment]
I've rebased and tested it on the newest master commit, is there anything left before this can be merged?

File: src/pages/StatusPage.vue
Comment (deefdragon) [inline src/pages/StatusPage.vue:81]: So, if someone is using tabbing to navigate the UI, having 2 separate buttons for the options may cause some strange behavior. Can you combine these into one button that determines what is shown/done based on the current state? 

Also, if possible, please add a tool tip that says "currently displaying tags/not displaying tags". Different UIs treat what the icon represents differently. IIRC Most use it as "make it go to this state" but some others do "it is currently in this state". The button text should cover it, but having a tool tip basically eliminates the ambiguity all together.

File: src/pages/StatusPage.vue
Comment (deefdragon) [inline src/pages/StatusPage.vue:499]:
```suggestion
                        return {
                            ...monitor,
```

File: src/pages/StatusPage.vue
Comment (deefdragon) [inline src/pages/StatusPage.vue:496]:
```suggestion
                return {
                    ...group,
```

File: server/routers/api-router.js
Comment (deefdragon) [inline server/routers/api-router.js:156]: Logically it looks pretty sound, but is there any chance you can get this to format a little cleaner? It was a bit of a pain to read through. 

- the if clause on 150 I kept misinterpreting the parens
- the SQL could be formatted for clarity
- the callback chaining could likely do with some newlines and indenting

Also, wont this `getSettings` for every `groupBean`? I don't think that is necessary.

**Review by deefdragon — CHANGES_REQUESTED** [review]
A few minor formatting things, one potential inefficiency, and some requests for the front end.

File: src/pages/StatusPage.vue
Comment (Fallstop) [reply src/pages/StatusPage.vue:81]: > having 2 separate buttons for the options may cause some strange behaviour

I also thought that at first glance, but it doesn't doesn't because `v-if` doesn't hide the element, it completely removes it from the DOM. `v-show` might cause this behaviour.

But yeah, I agree with combining the buttons, I was just trying to maintain consistency with the existing code-base, and the theme button is laid out like that. I wonder if I should also refactor the theme button to the same style while I'm at it?

> add a tool tip

Yup, great idea, but none of the other buttons have tool tips, might be a good idea to make that a separate PR?

> IIRC Most use it as "make it go to this state"

Yeah, the "make it go to this state" is much more common and accepted, but it is always good to clear ambiguity.

Here is the new combined button (tagsVisible is going to be a bool now):
```html
<button class="btn btn-secondary me-2" @click="changeTagsVisibilty(!tagsVisible)">
	<template v-if="tagsVisible">
		<font-awesome-icon icon="eye" />
		{{ $t("Hide Tags") }}
	</template>
	<template v-else>
		<font-awesome-icon icon="eye-slash" />
		{{ $t("Show Tags") }}
	</template>
</button>
```

File: server/routers/api-router.js
Comment (Fallstop) [reply server/routers/api-router.js]: > Also, wont this `getSettings` for every `groupBean`?

Good catch, I've cleaned up the formatting and loop in the latest commit so it should be quite a lot more readable.

File: src/pages/StatusPage.vue
Comment (deefdragon) [reply src/pages/StatusPage.vue]: I think it getting removed from the DOM would throw off the tab highlighting as there was nothing left _to_ highlight, which was the problem. Its fixed now regardless.

As for the tool-tips it's up to @louislam, but if he is ok with us having them, IMO we should add them right away, and then whenever touching old code (mind, only when reasonable, like with adding tests) instead of as one big PR.

File: src/pages/StatusPage.vue
Comment (chakflying) [inline src/pages/StatusPage.vue:493]:
```suggestion
            // On load, the status page will not include tags if it's not enabled for security reasons
```

**Review by chakflying — CHANGES_REQUESTED** [review]
Fix Grammar

File: server/routers/api-router.js
Comment (Saibamen) [inline server/routers/api-router.js:152]:
```suggestion
                monitorGroup.monitorList = await Promise.all(monitorGroup.monitorList.map(async (monitor) => {
```

File: server/routers/api-router.js
Comment (Saibamen) [inline server/routers/api-router.js:161]:
```suggestion
                    return {...monitor, tags: tags}
```

File: src/pages/StatusPage.vue
Comment (Saibamen) [inline src/pages/StatusPage.vue:496]:
```suggestion
            this.$root.publicGroupList = this.$root.publicGroupList.map((group) => {
```

File: src/pages/StatusPage.vue
Comment (Saibamen) [inline src/pages/StatusPage.vue:499]:
```suggestion
                    monitorList: group.monitorList.map((monitor) => {
```

**Pull request merged**
